File: python/cb-python-v4-python-v3.10-docker-official/context/couchbase-python-examples/simple-test.py
from datetime import timedelta


# needed for any cluster connection
from couchbase.auth import PasswordAuthenticator
from couchbase.cluster import Cluster
# needed for options -- cluster, timeout, SQL++ (N1QL) query, etc.
from couchbase.options import (ClusterOptions, ClusterTimeoutOptions,
                               QueryOptions)

# Update this to your cluster
endpoint = "--your-instance--.dp.cloud.couchbase.com"
username = "username"
password = "Password123!"
bucket_name = "travel-sample"
# User Input ends here.

# Connect options - authentication
auth = PasswordAuthenticator(username, password)

# Connect options - global timeout opts
timeout_opts = ClusterTimeoutOptions(kv_timeout=timedelta(seconds=10))

# get a reference to our cluster
cluster = Cluster('couchbases://{}'.format(endpoint),
                  ClusterOptions(auth, timeout_options=timeout_opts))

# Wait until the cluster is ready for use.
cluster.wait_until_ready(timedelta(seconds=5))

# get a reference to our bucket
cb = cluster.bucket(bucket_name)

cb_coll = cb.scope("inventory").collection("airline")


def upsert_document(doc):
    print("\nUpsert CAS: ")
    try:
        # key will equal: "airline_8091"
        key = doc["type"] + "_" + str(doc["id"])
        result = cb_coll.upsert(key, doc)
        print(result.cas)
    except Exception as e:
        print(e)

# get document function


def get_airline_by_key(key):
    print("\nGet Result: ")
    try:
        result = cb_coll.get(key)
        print(result.content_as[str])
    except Exception as e:
        print(e)

# query for new document by callsign


def lookup_by_callsign(cs):
    print("\nLookup Result: ")
    try:
        sql_query = 'SELECT VALUE name FROM `travel-sample`.inventory.airline WHERE callsign = $1'
        row_iter = cluster.query(
            sql_query,
            QueryOptions(positional_parameters=[cs]))
        for row in row_iter:
            print(row)
    except Exception as e:
        print(e)


airline = {
    "type": "airline",
    "id": 8091,
    "callsign": "CBS",
    "iata": None,
    "icao": None,
    "name": "Couchbase Airways",
}

upsert_document(airline)

get_airline_by_key("airline_8091")

lookup_by_callsign("CBS")

File: php/cb-php-v3-php-81-docker-official/context/couchbase-php-examples/simple-test.php
<?php

use \Couchbase\ClusterOptions;
use \Couchbase\Cluster;

// Update the connect string for your Couchbase on-prem or Capella instance.
$connectionString = "couchbases://cb.i-xxxxxxxxxxxxxx.cloud.couchbase.com?ssl=no_verify&timeout=10s";

// Update the credential values with your user credentials
$options = new \Couchbase\ClusterOptions();
$options->credentials("username", "password");

// get a cluster reference
$cluster = new \Couchbase\Cluster($connectionString, $options);

// get a bucket reference
$bucket = $cluster->bucket("travel-sample");

// get a user-defined collection reference$scope = $bucket->scope("tenant_agent_00");
$scope = $bucket->scope("tenant_agent_00");
$collection = $scope->collection("users");


/////
// Upsert / Get Example
/////

print("
------------------
upsert a user doc:
------------------
");
$upsertResult = $collection->upsert("user-doc-key", ["name" => "Ray", "points" => 499]);
print_r($upsertResult);

print("
-----------------
get the user doc:
-----------------
");
$getResult = $collection->get("user-doc-key");
print_r($getResult->content());


/////
// Query Example
/////

$queryString = "
SELECT
  a.id,
  a.name,
  a.icao
FROM
  default:`travel-sample`.inventory.airline a
WHERE
  a.country = \"United States\"
ORDER BY a.id DESC
LIMIT 5
";

printf("
---------------------
sample airline query:
---------------------
%s
---------------------
", $queryString);

$queryResult = $cluster->query($queryString);

// Iterate over the rows to access result data and print to the terminal.
foreach ($queryResult->rows() as $row) {
    print_r($row);
}

/////
// Ping Example
/////

print("
--------------------------
cluster ping all services:
--------------------------
");
$pingResult = $bucket->ping();
print_r($pingResult);

?>
